website/blog/2024/06-24-proxy-management-in-crawlee/index.md

```diff
@@ -0,0 +1,104 @@
+authorImageURL: https://avatars.githubusercontent.com/u/53312820?v=4
+authorTwitter: sauain
+---
+
+Hello Crawlee community,
+
+We are back with another blog, this time explaining how Crawlee rotates proxies and prevents crawlers from getting blocked.
+
+Proxies vary in quality, speed, reliability, and cost. There are a [few types of proxies](https://blog.apify.com/types-of-proxies/), such as datacenter and residential proxies. Datacenter proxies are cheaper but, on the other hand, more prone to getting blocked, and vice versa with residential proxies.
+
+It is hard for developers to decide which proxy to use while scraping data. We might get blocked if we use [datacenter proxies](https://blog.apify.com/datacenter-proxies-when-to-use-them-and-how-to-make-the-most-of-them/) for low-cost scraping, but residential proxies are sometimes too expensive for bigger projects. Developers need a system that can manage both costs and avoid getting blocked. To manage this, we recently introduced `TieredProxies` in Crawlee. Let’s take a look at it.
+
+:::note
+
+If you like reading this blog, we would be really happy if you gave [Crawlee a star on GitHub!](https://github.com/apify/crawlee/)
+
+:::
+
+## What are tiered proxies?
+
+Tiered proxies are a method of organizing and using different types of proxies based on their quality, speed, reliability, and cost. Tiered proxies allow you to rotate between a mix of proxy types to optimize your scraping activities.
+
+**Define proxy tiers**: You categorize your proxies into different tiers based on their quality. For example:
+
+- **High-tier proxies**: Fast, reliable, and expensive. Best for critical tasks where you need high performance.
+- **Mid-tier proxies**: Moderate speed and reliability. A good balance between cost and performance.
+- **Low-tier proxies**: Slow and less reliable but cheap. Useful for less critical tasks or high-volume scraping.
+
+## Features:
+
+- **Tracking errors**: The system monitors errors (e.g. failed requests, retries) for each domain.
+- **Adjusting tiers**: Higher-tier proxies are used if a domain shows more errors. Conversely, if a domain performs well with a high-tier proxy, the system will occasionally test lower-tier proxies. If successful, it continues using the lower tier, optimizing costs.
+- **Forgetting old errors**: Old errors are given less weight over time, allowing the system to adjust tiers dynamically as proxies' performance changes.
+
+### Working
+
+The `tieredProxyUrls` option in Crawlee's `ProxyConfigurationOptions` allows you to define a list of proxy URLs organized into tiers. Each tier represents a different level of quality, speed, and reliability.
+
+### Structure
+
+**Tiered Array**: Each element in the array is an object containing proxy URLs for that tier.
+
+### Usage
+
+**Fallback Mechanism**: Crawlee starts with the first tier of proxies. If proxies in the current tier fail, it will switch to the next tier.
+
+### Implementation:
```

Comment:
This is an example, not implementation

Reply:
also if it's a heading it shouldn't have the colon at the end I guess